compiler/passes/convert-uast.cpp

```diff
@@ -710,6 +710,12 @@ struct Converter {
         return remap;
       }
     } else {
+      if (name == USTR("single")) {
+        if (currentModuleType == MOD_USER) {
+          USR_WARN(node->id(), "'single' variables are deprecated");
```

Comment:
I'd probably adjust the message to suggest sync or atomic variables as an alternative.